_posts/2022-05-23-fine-grained-access-control-for-multitenancy.markdown

```diff
@@ -0,0 +1,369 @@
+---
+layout: post
+title: Fine Grained Access Control for Multi-tenancy
+description: A how to guide for architecting multitenant backend applications
+date: 2022-05-23
+author: Leon Ticharwa
+image: img/blog/multitenancy/multitenant.png
+tags: ['multitenant', 'typescript', 'tutorial', 'dynamodb', 'lambda', 'api gateway', 'authorization', 'authentication', 'apiGateway', 'jwt', 'json web token', 'cognito']
+---
+
+## What is Multi-tenancy
+
+Multi-tenancy is a software architecture that allows multiple users to access resources that reside in a shared computing environment. This architecture grants groups of users or individual users access to resources as defined by access boundaries put in place via the use of resource policies, IAM policies or some other similar mechanism. This concept is analogous to an apartment building where each tenant is given keys that only unlock their apartment door and other rooms within the apartment.
+
+Architecting multi-tenant cloud native software can be quite challenging due to the broad scope of technical know-how required to execute such an undertaking.
+What's even more challenging is finding cohesive guides/tutorials that take a top down approach to the somewhat elusive topic of multitenancy. This article is an attempt to produce a distilled "how-to" guide for those who may find themselves bogged down while building multi-tenant applications. You can clone the repo containing the reference code from here [reference code here](https://github.com/MechanicalRock/Multitenancy-AuthorizationAuthentication).
+
+## Some need to knows (Authorization and Authentication)
```

Comment:
The introductory information here is very useful, but is it detracting from the goal of this post - how to build a multi-tenant application?

Is it worth splitting this into two posts:
* An introductory post covering the various aspects of authentication and authorisation
* This post on multi-tenancy 
  * Refer to the terms you expect the reader to already be familiar with.  And link them to your other post for the detail.